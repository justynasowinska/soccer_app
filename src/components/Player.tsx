import * as React from 'react';
import { View, Text, StyleSheet } from 'react-native';

interface PropsType {
    id: number;
    x: number;
    y: number;
}

export const Player = (props: PropsType) => {
    const { id, x, y } = props;

    return (
        <View style={[styles.playerContainer, { top: x, left: y }]}>
            <View style={styles.player}>
                <Text style={styles.playerText}>{id}</Text>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    playerContainer: {
        position: 'absolute',
    },
    player: {
        justifyContent: 'center',
        alignItems: 'center',
        width: 50,
        height: 50,
        borderRadius: 100,
        backgroundColor: 'blue',
    },
    playerText: {
        fontSize: 12,
        color: 'white'
    }
});