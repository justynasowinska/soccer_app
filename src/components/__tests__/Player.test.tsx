import * as React from 'react';
import { shallow } from 'enzyme';

import { Player } from '../Player';

describe('Player', () => {
    it('matches to the snapshot', () => {
        const component = shallow(
            <Player
                id={0}
                x={0.434}
                y={0.334}
            />
        );

        expect(component).toMatchSnapshot();
    });
});