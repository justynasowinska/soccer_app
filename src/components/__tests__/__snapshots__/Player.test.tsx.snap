// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player matches to the snapshot 1`] = `
<AnimatedComponent
  style={
    Array [
      Object {
        "position": "absolute",
      },
      Object {
        "bottom": "33.4%",
        "left": "43.4%",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#00B4CC",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 35,
        "justifyContent": "center",
        "width": 35,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00B4CC",
          "borderRadius": 100,
          "height": "80%",
          "justifyContent": "center",
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F6F6F6",
            "fontSize": 14,
          }
        }
      >
        0
      </Text>
    </View>
  </View>
</AnimatedComponent>
`;
